apis/r/R/SOMAExperimentAxisQuery.R

```diff
@@ -994,6 +1132,233 @@ SOMAExperimentAxisQuery <- R6::R6Class(
       }
       return(mat)
     },
+    # Helper methods to load aspects of a measurement for the ecosystems
+    .load_df = function(df_name = c('obs', 'var'), index = NULL, column_names = NULL) {
+      stopifnot(
+        is.character(df_name),
+        is.null(index) || is_scalar_character(index),
+        is.null(column_names) || is.character(column_names) || is_scalar_logical(column_names)
+      )
+      df_name <- match.arg(arg = df_name)
+      switch(
+        EXPR = df_name,
+        obs = {
+          soma_df <- self$obs_df
+          soma_reader <- self$obs
+          soma_joinids <- self$obs_joinids()
+        },
+        var = {
+          soma_df <- self$var_df
+          soma_reader <- self$var
+          soma_joinids <- self$var_joinids()
+        }
+      )
+      ids <- if (is.null(index)) {
+        paste0(df_name, soma_joinids$as_vector())
+      } else {
+        index <- match.arg(arg = index, choices = soma_df$attrnames())
+        soma_reader(index)$concat()$GetColumnByName(index)$as_vector()
```

Comment:
Looks like we're performing 2 separate reads here: once to retrieve ids from the designated index column and the once to read in the rest of the data. Can we consolidate to avoid the extra overhead?